### ISSUE TITLE
Support torchScript in DGL sparse library

### ISSUE THREAD

**Issue by paoxiaode** [dmlc/dgl#5275]
## 🔨Work Item

**IMPORTANT:**
* This template is only for dev team to track project progress. For feature request or bug report, please use the corresponding issue templates.
* DO NOT create a new work item if the purpose is to fix an existing issue or feature request. We will directly use the issue in the project tracker.

Project tracker: https://github.com/orgs/dmlc/projects/2

## Description

To support the torchScript in DGL sparse library,   DGL sparse API need some change (like https://github.com/dmlc/dgl/commit/be3b0e6127446c26d4d157d5746f9e82e3a7cc02).

**Comment by wondey-sh** [comment]
Hi @frozenbugs , just curious to know whether the team has any concrete plan to support TorchScript with Sparse API? Sparse API works well in my projects, but the lack of TorchScript support is definitely an obstacle to the deployment. Thanks a lot.

**Comment by jermainewang** [comment]
Hi @wondey-sh , it looks like TorchScript will be superseded by TorchDynamo in the future, so we put this feature request on hold.  For TorchDynamo compatibility, we are blocked by the lack of support for custom class (which we rely on for defining the SparseMatrix class). We would love to hear if the community has any solution or workaround for that.

**Comment by zhouxiaozhang** [comment]
> Hi @frozenbugs , just curious to know whether the team has any concrete plan to support TorchScript with Sparse API? Sparse API works well in my projects, but the lack of TorchScript support is definitely an obstacle to the deployment. Thanks a lot.



> Hi @frozenbugs , just curious to know whether the team has any concrete plan to support TorchScript with Sparse API? Sparse API works well in my projects, but the lack of TorchScript support is definitely an obstacle to the deployment. Thanks a lot.



> Hi @frozenbugs , just curious to know whether the team has any concrete plan to support TorchScript with Sparse API? Sparse API works well in my projects, but the lack of TorchScript support is definitely an obstacle to the deployment. Thanks a lot.

hi,@wondey-sh
May I ask if there is any other solution for deploying DGL online in your project now?
